pydantic/json_schema.py

```diff
@@ -742,13 +742,51 @@ def literal_schema(self, schema: core_schema.LiteralSchema) -> JsonSchemaValue:
         elif types == {int}:
             result['type'] = 'integer'
         elif types == {float}:
-            result['type'] = 'number'
+            result['type'] = 'numeric'
```

Comment:
we have a weird mixture of `numeric` and `number in this file, I changed this to match `enum_schema`, which had to use `numeric` to match existing tests.